src/types.rs

```diff
@@ -519,6 +594,20 @@ impl dyn Interface {
         )?)?)
     }
 
+    pub fn get_data_wasmv1<T: DeserializeOwned>(
+        &self,
+        key: &[u8],
+        address: Option<NativeAddress>,
+    ) -> Result<T> {
+        Ok(serde_json::from_str::<T>(std::str::from_utf8(
```

Comment:
let's keep it like this for the moment,
I need to think twice about it
but I have the feeling that we are doing too much conversion here. 
serializing data with protobuf is probably enough

Reply:
Added a TODO comment